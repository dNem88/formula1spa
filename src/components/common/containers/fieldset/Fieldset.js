import React from 'react'
import styles from './fieldset.module.css'

 function Fieldset({size, content, color, children}) {

    return (
         <fieldset className={styles[`fieldset-${size}`]}>
                <legend className={styles[`legend-${color}`]}>{content}</legend>
                {children}
        </fieldset>
    )

}

export default Fieldset