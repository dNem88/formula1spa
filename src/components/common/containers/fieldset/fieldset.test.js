import React from 'react'
import renderer from 'react-test-renderer'
import Fieldset from './Fieldset'
import {screen, render} from '@testing-library/react'

const mockProps = {
    size: 'full',
    content: 'test-content',
    color: 'test-color',
    children: 'CHILDREN'
}
describe('Testing Fieldset Component', () => {
    it('render properly based on props', () => {
        const snapshot = renderer.create(
            <Fieldset {...mockProps}/>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.props.className).toBe('fieldset-full')
        expect(snapshot.type).toBe('fieldset')
        expect(snapshot.children[1]).toBe('CHILDREN')
    })
    test('Tags have correct values', () => {
        render(<Fieldset {...mockProps}/>)
        expect(screen.getByText(mockProps.content)).toBeTruthy()
        expect(screen.getByText('test-content')).toHaveClass('legend-test-color')
        
    })
})