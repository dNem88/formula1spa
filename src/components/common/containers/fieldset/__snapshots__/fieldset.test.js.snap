// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Fieldset Component render properly based on props 1`] = `
<fieldset
  className="fieldset-full"
>
  <legend
    className="legend-test-color"
  >
    test-content
  </legend>
  CHILDREN
</fieldset>
`;
